truckersmp_cli/__init__.py

```diff
@@ -143,6 +146,70 @@ def data(self):
         return self._data
 
 
+def perform_self_update():
+    """
+    Update files to latest release. Do nothing for Python package.
+
+    This function checks the latest GitHub release first.
+    If local version is not up-to-date, this function retrieves the URL to latest
+    GitHub release asset (.tar.xz) and replaces existing files with extracted files.
+    """
+    # we don't update when Python package is used
+    try:
+        with open(os.path.join(os.path.dirname(Dir.scriptdir), "RELEASE")) as f:
+            current_release = f.readline().rstrip()
+    except Exception:
+        sys.exit("'RELEASE' file doesn't exist. Self update aborted.")
+
+    # get latest release
+    logging.info("Retrieving RELEASE from master")
+    try:
+        with urllib.request.urlopen(URL.release) as f:
+            release = f.readline().rstrip().decode("ascii")
+    except Exception as e:
+        sys.exit("Failed to retrieve RELEASE file: {}".format(e))
+
+    # do nothing if the installed version is latest
+    if release == current_release:
+        logging.info("Already up-to-date.")
+        return
+
+    # retrieve the release asset
+    archive_url = URL.rel_tarxz_tmpl.format(release)
+    logging.info("Retrieving release asset {}".format(archive_url))
+    try:
+        with urllib.request.urlopen(archive_url) as f:
+            asset_archive = f.read()
+    except Exception as e:
+        sys.exit("Failed to retrieve release asset file: {}".format(e))
+
+    # unpack the archive
+    logging.info("Unpacking archive {}".format(archive_url))
+    topdir = os.path.dirname(Dir.scriptdir)
+    try:
+        with tarfile.open(fileobj=io.BytesIO(asset_archive), mode="r:xz") as f:
+            f.extractall(topdir)
+    except Exception as e:
+        sys.exit("Failed to unpack release asset file: {}".format(e))
+
+    # update files
+    archive_dir = os.path.join(topdir, "truckersmp-cli-" + release)
+    for root, dirs, files in os.walk(archive_dir, topdown=False):
```

Comment:
Actually I was aware of this and it was intentionally untouched because I often forget how to use `os.walk()`. :sweat_smile:
Is your IDE satisfied with `_dirs`? Otherwise I would forget what is the 2nd element. :worried:

Reply:
It is! It's like magic.
I think it's using pylint in the background.